ecs-agent/acs/session/session.go

```diff
@@ -253,17 +255,26 @@ func (s *session) startSessionOnce(ctx context.Context) error {
 
 	// Invoke Connect method as soon as we create client. This will ensure all the
 	// request handlers to be associated with this client have a valid connection.
+	acsConnectionStartTime := time.Now()
 	disconnectTimer, err := client.Connect(metrics.ACSDisconnectTimeoutMetricName, s.disconnectTimeout,
 		s.disconnectJitter)
+
+	// Metric created for determining whether ACS connection is successful or not
+	s.metricsFactory.New(metrics.ACSSessionCallName).Done(err)
 	if err != nil {
 		logger.Error("Failed to connect to ACS", logger.Fields{
 			"containerInstanceARN": s.containerInstanceARN,
 			field.Error:            err,
 		})
 		return err
 	}
+	s.metricsFactory.New(metrics.ACSSessionCallDurationName).WithGauge(time.Since(acsConnectionStartTime)).Done(nil)
 	defer disconnectTimer.Stop()
 
+	if s.GetFirstACSConnectionTime().IsZero() {
```

Comment:
I'm not sure about what the purpose of this timestamp is here. Looks whenever the agent is restarted for some reason, this timestamp will be updated. Will this be a problem?

Reply:
+1 (not sure if this was discussed offline or not)